ur_robot_driver/src/ros/hardware_interface.cpp

```diff
@@ -318,6 +320,18 @@ bool HardwareInterface::init(ros::NodeHandle& root_nh, ros::NodeHandle& robot_hw
   ur_driver_->startRTDECommunication();
   ROS_INFO_STREAM_NAMED("hardware_interface", "Loaded ur_robot_driver hardware_interface");
 
+  // Setup the mounted payload through a ROS service
+  set_payload_srv_ = robot_hw_nh.advertiseService<ur_msgs::SetPayload::Request, ur_msgs::SetPayload::Response>(
+      "set_payload", [&](ur_msgs::SetPayload::Request& req, ur_msgs::SetPayload::Response& resp) {
+        std::stringstream cmd;
+        cmd << "sec setup():" << std::endl
+            << " set_payload(" << req.payload << ", [" << req.center_of_gravity.x << ", " << req.center_of_gravity.y
+            << ", " << req.center_of_gravity.z << "])" << std::endl
```

Comment:
Is relying on `<<` to convert to `string` locale safe?

Could end up with multiple commas in the string, which would probably upset UR's parser.

Reply:
I did not think about that. I would never have though about the stream operator using locales in the background.

In a quick example I could not get a stream into locale-like formatting without explicitly setting the locale of the stream. However, to make sure, we could set the string's locale explicitly to [`classic`](https://en.cppreference.com/w/cpp/locale/locale/classic)

Reply:
I just want to avoid another ros/urdfdom_headers#41.

If stream io doesn't suffer from this, all the better.